# AXES
# 0 = Left Stick X
# 1 = Left Stick Y
# 2 = Right Stick X
# 3 = Right Stick Y
# 4 = Left Trigger
# 5 = Right Trigger

# BUTTONS
# 0   = Face Bottom
# 1   = Face Right
# 2   = Face Left
# 3   = Face Top
# 4   = Back Button
# 5   = PS5
# 6   = Menu Button
# 7   = Left Stick Button
# 8   = Right Stick Button
# 9   = Left Shoulder
# 10  = Right Shoulder
# 11  = DPad up
# 12  = DPad down
# 13  = DPad left
# 14  = DPad right

gamepad_node:
  ros__parameters:
    button_mapping:
      move_home: 3
      move_sleep: 1
      dead_man_switch: 10
      wrist_rotation_left: 7
      wrist_rotation_right: 8
      switch_controller: 6  # Menu button for switching controllers
    axis_mapping:
      left_joystick:
        x: 0
        y: 1
      right_joystick:
        x: 2
        y: 3
      triggers:
        left: 4
        right: 5

    controllers:
      joint_trajectory_controller:
        whitelisted: true
      freedrive_controller:
        whitelisted: true
